pkg/service/scheduler/service_integration_test.go

```diff
@@ -991,4 +992,101 @@ func TestServiceScheduleIntegration(t *testing.T) {
 			t.Fatal("Expected resumed")
 		}
 	})
+
+	t.Run("task out of maintenance window", func(t *testing.T) {
+		h := newSchedTestHelper(t, session)
+		defer h.close()
+		ctx := context.Background()
+
+		// Create ad-hoc window.
+		// Watch out for the time near the midnight.
+		stop := now().Add(15 * time.Second)
+		start := stop.Add(15 * time.Second)
+		if start.Hour() < stop.Hour() {
+			time.Sleep(2 * time.Minute)
+			stop = now().Add(15 * time.Second)
+			start = stop.Add(15 * time.Second)
+		}
+
+		timeToDuration := func(t time.Time) time.Duration {
+			return time.Hour*time.Duration(t.Hour()) +
+				time.Minute*time.Duration(t.Minute()) +
+				time.Second*time.Duration(t.Second())
+		}
+
+		window := []scheduler.WeekdayTime{
+			{WeekdayTime: sched.WeekdayTime{Weekday: sched.EachDay, Time: 0}},
+			{WeekdayTime: sched.WeekdayTime{Weekday: sched.EachDay, Time: timeToDuration(stop)}},
+			{WeekdayTime: sched.WeekdayTime{Weekday: sched.EachDay, Time: timeToDuration(start)}},
+			{WeekdayTime: sched.WeekdayTime{Weekday: sched.EachDay, Time: 24*time.Hour - time.Second}},
+		}
+
+		Print("When: task is scheduled")
+		task := h.makeTask(scheduler.Schedule{
+			StartDate: now(),
+			Window:    window,
+		})
+		if err := h.service.PutTask(ctx, task); err != nil {
+			t.Fatal(err)
+		}
+
+		Print("Then: task runs")
+		h.assertStatus(task, scheduler.StatusRunning)
+
+		Print("Then: task goes out of window")
+		WaitCond(h.t, func() bool {
+			v := h.getStatus(task)
+			return v == scheduler.StatusWaiting
+		}, _interval, 2*time.Minute)
+
+		Print("Then: task starts in the next window")
+		WaitCond(h.t, func() bool {
+			v := h.getStatus(task)
+			return v == scheduler.StatusRunning
+		}, _interval, 2*time.Minute)
+
+		h.runner.Done()
+		Print("Then: task stops with the status done")
+		h.assertStatus(task, scheduler.StatusDone)
+	})
+
+	t.Run("task ends with context error", func(t *testing.T) {
```

Comment:
Just for my understanding.
This test validates the scenario where the task is running, but one of the operations timed out and it returned "context canceled", right ?

This test bypasses / mocks a bit.
Do you see any possibility of validating it bit more extensive way ? 
Maybe SCT ?
I'm ok with the current tests, but I'm afraid that something may be missed still.

Reply:
> Just for my understanding.
This test validates the scenario where the task is running, but one of the operations timed out and it returned "context canceled", right ?

That's right. I also validated that this test was failing prior to introducing dedicated scheduler errors.

> This test bypasses / mocks a bit.
Do you see any possibility of validating it bit more extensive way ?
Maybe SCT ?

I think that we don't have full integration tests in SM repo, only integration tests at service level.
I'm not sure if it's easy to do with SCT as well (I'm afraid of such test's flakiness).

>  I'm ok with the current tests, but I'm afraid that something may be missed still.

I went over all places in the code that used `context.Canceled` and `context.DeadlineExceeded` (there weren't much of them), and applied the new logic there as well. Note, that all the tasks still return context errors like before. The only thing that's different is how we check if task ended with an error or because of scheduler interruption, so the changes are not that broad.
I would say that tests on service level are good enough in this case.

Reply:
It's fine. The PR is approved.